### PR TITLE
PEG-4936 migrate to pleo gradle plugin

### PR DESCRIPTION
**Pull request by andriygg**
Linear story: [PEG-4936](https://linear.app/pleo/issue/PEG-4936)

Have you done ...
* [ ] Unit tests

### CODE REVIEW

**Review by javiergarciapleo — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -102,85 +77,12 @@ subprojects {
...
-                            name = 'Pleo'
-                            email = 'dev@pleo.io'
-                        }
-                    }
-                    scm {
-                        connection = 'scm:git:git@github.com:pleo-io/prop.git'
-                        developerConnection = 'scm:git:git@github.com:pleo-io/prop.git'
-                        url = 'https://github.com/pleo-io/prop'
-                    }
-                }
```
Comment (erwinw) [inline build.gradle:133]: Where is the POM info now configured? I can't find it in your PR, but maybe I've overlooked it?

**Review by erwinw — COMMENTED** [review]
That's a lot of leg-work, thanks so much!

I wonder if the publication to Maven still works with this?

File: build.gradle [hunk 2]
```diff
@@ -102,85 +77,12 @@ subprojects {
...
-    reportPath = "build/reports/jacoco/jacocoAggregatedReport/jacocoAggregatedReport.xml"
-    reportSourceSets += subprojects.sourceSets.main.allSource.srcDirs.flatten()
+tasks.check {
+    finalizedBy("formatKotlin")
 }
 
-tasks.named('coverallsJacoco') {
-    dependsOn jacocoAggregatedReport
+pleoRepositoryExtension {
+    publish.set(subprojects.toSet())
```
Comment (andriygg) [inline build.gradle:87]: @erwinw AFAIK, pleo-gradle-plugin is responsible for a new artifact publishing flow, so it has pom definitions there... that's why I've removed those according to [migration docs](https://www.notion.so/pleo/Adopting-the-Pleo-Gradle-plugin-78834e6e171e458197db1773d9242894?pvs=4#f213d1d2cfe4498c86e023c7329379d4)

File: build.gradle (hunk 1 above)
Comment (andriygg) [reply build.gradle:133]: yeah, I"ve checked it locally and all artifacts have been successfully published to the local repo, with sources as well

Comment (erwinw) [reply build.gradle:133]: Just to be 200% explicit: we need to publish to Maven central for this repo. You're convinced you can do this and update [this entry](https://central.sonatype.com/artifact/io.pleo/prop-all)?

File: build.gradle (hunk 2 above)
Comment (erwinw) [reply build.gradle:87]: I'm not convinced that's the "right type of publishing"; AFAIK that's only to our private Github package registry, not Maven Central?

File: build.gradle (hunk 1 above)
Comment (andriygg) [reply build.gradle:133]: thank you for pointing, I haven't noticed that prop is published to maven central

Comment (andriygg) [reply build.gradle]: ok, I"ve partially reverted back my changes... one thing I'm not sure, what password to use for publishing: old `GITHUB_TOKEN` or `GRADLE_WRITE_KEY`... for now I've left `GITHUB_TOKEN` going to check if it still works

File: build.gradle (hunk 2 above)
Comment (andriygg) [reply build.gradle]: removed this task

File: build.gradle (hunk 1 above)
Comment (andriygg) [reply build.gradle]: ok, we cannot use Pleo-gradle-plugin for public repos for now see [DEVX-1972](https://linear.app/pleo/issue/DEVX-1972/hey-ive-spotted-the-following-issue-[when-running-ci)

File: build.gradle [hunk 3]
```diff
@@ -184,3 +191,7 @@ coverallsJacoco {
 tasks.named('coverallsJacoco') {
     dependsOn jacocoAggregatedReport
 }
+
+//tasks.named('check') {
```
Comment (andriygg) [inline build.gradle:195]: @erwinw I've been struggling to add support of kotlinter pre [commit push hook](https://github.com/jeremymailen/kotlinter-gradle?tab=readme-ov-file#git-hooks) and haven't succeeded yet. If you would have some time to take a look, I'd appreciate 🙏

File: prop-all/src/test/kotlin/io/pleo/prop/PropTest.kt [hunk 4]
```diff
@@ -40,19 +40,21 @@ import javax.sql.DataSource
 class PropTest {
     @Test
     fun `can read complex properties`() {
-        val injector = createInjector(
-            Module { binder: Binder ->
-                binder.bind(ComplexObjects::class.java)
-            }
-        )
+        val injector =
```
Comment (andriygg) [inline prop-all/src/test/kotlin/io/pleo/prop/PropTest.kt:43]: manually applied linter formatting

**Comment by erwinw** [comment]
It seems like [the assign random codeowners step does not work](https://github.com/pleo-io/prop/actions/runs/7708929486/job/21009017674?pr=232#step:1:35) because it's a public repo that does not have access to the private [pleo-io/assign-random-codeowners](https://github.com/pleo-io/assign-random-codeowners) repo, @andriygg .

**Comment by andriygg** [comment]
> It seems like [the assign random codeowners step does not work](https://github.com/pleo-io/prop/actions/runs/7708929486/job/21009017674?pr=232#step:1:35) because it's a public repo that does not have access to the private [pleo-io/assign-random-codeowners](https://github.com/pleo-io/assign-random-codeowners) repo, @andriygg .

I've removed this workflow as well. But all other work for now

**Review by erwinw — APPROVED** [review]
LGTM - thanks so much!

**Comment by andriygg** [comment]
@erwinw would you have some time to look at configuration of the kotlinter precommit hook? I've spent some amount of time and still cannot register it for the root project

**Comment by erwinw** [comment]
> @erwinw would you have some time to look at configuration of the kotlinter precommit hook? I've spent some amount of time and still cannot register it for the root project

Would [this approach](https://github.com/pleo-io/dysnomia/commit/2ecaa3ea44e254959572221f37b91265569cc911) work?

**Comment by andriygg** [comment]
> > @erwinw would you have some time to look at configuration of the kotlinter precommit hook? I've spent some amount of time and still cannot register it for the root project
> 
> Would [this approach](https://github.com/pleo-io/dysnomia/commit/2ecaa3ea44e254959572221f37b91265569cc911) work?

That is right approach for sure, but the problem is that I'm getting error
```sh
A problem occurred evaluating root project 'prop'.
> Task with name 'check' not found in root project 'prop'.
```

when include depends on for check task
```gradle
tasks.named('check') {
    dependsOn 'installKotlinterPrePushHook'
}
```

I've tried a few approached to fix it up, but looks like I'm missing something. So that's why I'm asking if you'd have some time to take a look. It should be something trivial

**Comment by erwinw** [comment]
> > > @erwinw would you have some time to look at configuration of the kotlinter precommit hook? I've spent some amount of time and still cannot register it for the root project
> 
> > 
> 
> > Would [this approach](https://github.com/pleo-io/dysnomia/commit/2ecaa3ea44e254959572221f37b91265569cc911) work?
> 
> 
> 
> That is right approach for sure, but the problem is that I'm getting error
> 
> ```sh
> 
> A problem occurred evaluating root project 'prop'.
> 
> > Task with name 'check' not found in root project 'prop'.
> 
> ```
> 
> 
> 
> when include depends on for check task
> 
> ```gradle
> 
> tasks.named('check') {
> 
>     dependsOn 'installKotlinterPrePushHook'
> 
> }
> 
> ```
> 
> 
> 
> I've tried a few approached to fix it up, but looks like I'm missing something. So that's why I'm asking if you'd have some time to take a look. It should be something trivial

Ah I see. Sure, I'll have a look!

**Pull request merged**